log4j-api/src/main/java/org/apache/logging/log4j/CloseableThreadContext.java

```diff
@@ -128,6 +153,37 @@ public Instance put(final String key, final String value) {
         }
 
         /**
+         * Populates the Thread Context Map with the supplied key/value pairs. Any existing keys in the
+         * {@link ThreadContext} will be replaced with the supplied values, and restored back to their original value when
+         * the instance is closed.
+         *
+         * @param values The map of key/value pairs to be added
+         * @return a new instance that will back out the changes when closed.
+         * @since 2.7.1
+         */
+        public Instance putAll(final Map<String, String> values) {
```

Comment:
I don't think that's sensible here;  the `CloseableThreadContext` will backup any existing values and restore them (or remove them, if no existing values) when closed. So even if we do use `ThreadContext.putAll()` it will still be necessary to iterate over the `Map`.